src/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -241,13 +241,8 @@ public void Set(int index, bool value)
         public void SetAll(bool value)
         {
             int fillValue = value ? -1 : 0;
-            int[] array = m_array;
-
-            for (int i = 0; i < array.Length; i++)
-            {
-                array[i] = fillValue;
-            }
-
+            int arrayLength = GetInt32ArrayLengthFromBitLength(Length);
+            m_array.AsSpan(0, arrayLength).Fill(fillValue);
```

Comment:
@ahsonkhan had previously looked into doing this and found there were regressions for smaller inputs.  See https://github.com/dotnet/corefx/pull/33367#discussion_r233164307 for a discussion with him and @GrabYourPitchforks.

Reply:
I expected that, but the perf results for a single "int" (32-bit) "SetAll" still showed an improvement, which is surprising:

| System.Collections.Tests.Perf_BitArray.BitArraySetAll(Size: 32)             |      1.17 |             4.87 |             4.14 |         |

Reply:
I should probably run them again, and try harder to get a quiescent machine.